### ISSUE TITLE
Question: Are module-level docstrings supported?

### ISSUE THREAD

**Issue by hacker-DOM** [microsoft/pylance-release#2551]
Hi there!

When I have a module (file) `f.py` with

```python
__doc__ = "xyz"
```

And in a different file

```python
import my_package.f
```

i would expect that triggering `editor.action.showHover` would show "xyz".

However, it only shows "(module) f".

Thank you for considering my issue!

**Comment by erictraut** [comment]
Module-level docstrings are supported if you have a docstring at the top of the module. They are not supported if you dynamically assign a value to the `__doc__` variable. Keep in mind that a static type analyzer doesn't actually run the code.

**Comment by hacker-DOM** [comment]
Thank you for your answer, however, I am still getting unexpected behavior. I have tried to [document this](https://github.com/hacker-DOM/pylance-definition-preview).

tldr: it only works for relative imports, not for absolute imports

Thanks !

**Comment by erictraut** [comment]
Docstrings are displayed for symbols, not for import paths. In the expression `from a.b.c import d`, `a.b.c` is an import path (as is `a` and `a.b`); `d` is the imported symbol. That explains the difference you're seeing.

It sounds like you are interested in seeing module docstrings for module paths when they refer to a module. That's a reasonable enhancement request.

**Comment by judej** [comment]
Moving this issue to discussion as an enhancement request for comments and upvotes.
